pkg/collections.go

```diff
@@ -0,0 +1,264 @@
+						return nil
+					}, backoff.WithContext(backoff.WithMaxRetries(backoff.NewExponentialBackOff(), uint64(backoffMaxTries)), ctx)); err != nil {
+						c.Logger.Error().Err(err).Str("collection", sourceColl.Name()).Msg("Backoff eventually failed.")
+						cursor.Close()
+						return err
+					}
+				}
+
+				if !cursor.HasMore() {
+					break
+				}
+			}
+			cursor.Close()
+
+			// Copy over all indexes for this collection.
+			if err := c.copyIndexes(restoreCtx, sourceColl, destinationColl); err != nil {
+				c.Logger.Error().Err(err).Str("collection", sourceColl.Name()).Msg("Failed to copy all indexes.")
+				return err
+			}
+
+			return nil
+		})
+	}
+	if err := g.Wait(); err != nil {
+		log.Error().Err(err).Msg("One of the workers failed to copy data.")
+		return err
+	}
+	s.Stop()
+	log.Debug().Str("source-database", sourceDB.Name()).Msg("Done copying database data.")
+	return nil
+}
+
+// copyIndexes copies all indexes for a collection to destination collection.
+func (c *copier) copyIndexes(ctx context.Context, sourceColl driver.Collection, destinationColl driver.Collection) error {
+	var indexes []driver.Index
+	c.backoffCall(ctx, func() error {
+		idxs, err := sourceColl.Indexes(ctx)
+		if err != nil {
+			return err
+		}
+		indexes = idxs
+		return nil
+	})
+	for _, index := range indexes {
+		c.backoffCall(ctx, func() error {
+			switch index.Type() {
+			case driver.TTLIndex:
+				var field string
+				if len(index.Fields()) > 0 {
+					field = index.Fields()[0]
+				}
+				if _, _, err := destinationColl.EnsureTTLIndex(ctx, field, index.ExpireAfter(), &driver.EnsureTTLIndexOptions{
+					Name: index.UserName(),
+				}); err != nil {
+					return err
+				}
+			case driver.PersistentIndex:
+				if _, _, err := destinationColl.EnsurePersistentIndex(ctx, index.Fields(), &driver.EnsurePersistentIndexOptions{
+					Name:   index.UserName(),
+					Unique: index.Unique(),
+					Sparse: index.Sparse(),
+				}); err != nil {
+					return err
+				}
+			case driver.SkipListIndex:
+				if _, _, err := destinationColl.EnsureSkipListIndex(ctx, index.Fields(), &driver.EnsureSkipListIndexOptions{
+					Name:          index.UserName(),
+					Unique:        index.Unique(),
+					Sparse:        index.Sparse(),
+					NoDeduplicate: !index.Deduplicate(),
+				}); err != nil {
+					return err
+				}
+			case driver.HashIndex:
+				if _, _, err := destinationColl.EnsureHashIndex(ctx, index.Fields(), &driver.EnsureHashIndexOptions{
+					Name:          index.UserName(),
+					Unique:        index.Unique(),
+					Sparse:        index.Sparse(),
+					NoDeduplicate: !index.Deduplicate(),
+				}); err != nil {
+					return err
+				}
+			case driver.FullTextIndex:
+				if _, _, err := destinationColl.EnsureFullTextIndex(ctx, index.Fields(), &driver.EnsureFullTextIndexOptions{
+					Name:      index.UserName(),
+					MinLength: index.MinLength(),
+				}); err != nil {
+					return err
+				}
+			case driver.GeoIndex:
```

Comment:
There are multiple GEO index types

Reply:
What do you mean? I can only create this one. I don't see any more options in driver for different GeoIndexes.